emukit/quadrature/kernels/integration_measures.py

```diff
@@ -0,0 +1,147 @@
+        """
+        Meaningful box-bounds around the measure. Outside this box, the measure should be virtually zero.
+
+        :return: box in which the measure lies. List of D tuples, where D is the dimensionality and the tuples contain
+        the lower and upper bounds of the box i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)]
+        """
+        raise NotImplementedError
+
+
+class UniformMeasure(IntegrationMeasure):
+    """The Uniform measure"""
+
+    def __init__(self, bounds: List[Tuple[float, float]]):
+        """
+        :param bounds: List of D tuples, where D is the dimensionality of the domain and the tuples contain the lower
+        and upper bounds of the box defining the uniform measure i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)]
+        """
+        super().__init__('UniformMeasure')
+
+        # checks if lower bounds are smaller than upper bounds.
+        for bounds_d in bounds:
+            lb_d, ub_d = bounds_d
+            if lb_d >= ub_d:
+                raise ValueError("Upper bound of uniform measure must be larger than lower bound. Found a pair "
+                                 "containing (" + str(lb_d) + ", " + str(ub_d) + ").")
+
+        self.bounds = bounds
+        # uniform measure has constant density which is computed here.
+        self.density = self._compute_density()
+
+    def _compute_density(self) -> float:
+        differences = np.array([x[1] - x[0] for x in self.bounds])
+        volume = np.prod(differences)
+
+        if not volume > 0:
+            raise NumericalPrecisionError("Domain volume of uniform measure is not positive. Its value is "
+                                          + str(volume) + ". It might be numerical problems...")
+        return np.float(1. / volume)
+
+    def compute_density(self, x: np.ndarray) -> np.ndarray:
+        """
+        Computes the density at point x
+        :param x: points at which density is computes, shape (num_points, dim)
+        :return: the density at x, shape (num_points, )
+        """
+        # check if points are inside the box
+        bounds_lower = np.array([b[0] for b in self.bounds])
+        bounds_upper = np.array([b[1] for b in self.bounds])
+        inside_lower = 1 - (x < bounds_lower)
+        inside_upper = 1 - (x > bounds_upper)
+        inside_upper_lower = (inside_lower * inside_upper).sum(axis=1) == x.shape[1]
+        return inside_upper_lower * self.density
+
+    def get_box(self) -> List[Tuple[float, float]]:
+        """
+        Meaningful box-bounds around the measure. Outside this box, the measure should be virtually zero.
+
+        :return: box in which the measure lies. List of D tuples, where D is the dimensionality and the tuples contain
+        the lower and upper bounds of the box i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)]
+        """
+        return self.bounds
+
+
+class IsotropicGaussianMeasure(IntegrationMeasure):
+    """The isotropic Gaussian measure"""
```

Comment:
The previous measure seems clear and simple, this imho requires a reference, maybe a link or a more detailed explanation of what is this object

Reply:
I am not sure what I should add. It is an isotropic Gaussian with an isotropic Gaussian density. That's a pretty well defined term. The description of the variance in the docstring says in addition that the variance is scalar.  I could add a formula if you think that would help.

Reply:
a formula would definitely help. imagine a person who never saw term before. is there a way to write a docstring so that they wouldn't need to go to wiki while reading this code?

Reply:
OK, I'll add a formula as well as this phrase: " An isotropic Gaussian is a Gaussian with scalar co-variance matrix." 

Just for context, the measure is part of the problem/task. So it is a quantity that is designed by, and most likely well understood by the user.